test/12_conditional_commit.ts

```diff
@@ -0,0 +1,1056 @@
+        );
+
+        const tokenSupplyKey = eventArgs._tokenIdSupply;
+
+        const {txValue, DEPOSIT, PRICE} = await generateInputs(
+          users.buyer,
+          constants.buyer_deposit,
+          constants.product_price
+        );
+
+        const vDeposit = DEPOSIT.v;
+        const rDeposit = DEPOSIT.r;
+        const sDeposit = DEPOSIT.s;
+        const vPrice = PRICE.v;
+        const rPrice = PRICE.r;
+        const sPrice = PRICE.s;
+
+        const buyerInstance = contractBosonRouter.connect(
+          users.other1.signer
+        ) as BosonRouter;
+        await expect(
+          buyerInstance.requestVoucherTKNTKNWithPermit(
+            tokenSupplyKey,
+            users.seller.address,
+            txValue,
+            deadline,
+            vPrice,
+            rPrice,
+            sPrice,
+            vDeposit,
+            rDeposit,
+            sDeposit
+          )
+        ).to.be.revertedWith(revertReasons.NOT_ELIGIBLE);
+      });
+    }); // end TKNTKN
+
+    describe('TKNTKN same', () => {
+      beforeEach(async () => {
+        await deployContracts();
+
+        timestamp = await Utils.getCurrTimestamp();
+        constants.PROMISE_VALID_FROM = timestamp;
+        constants.PROMISE_VALID_TO = timestamp + 2 * constants.SECONDS_IN_DAY;
+
+        const tokensToMint = BN(constants.product_price).mul(
+          BN(constants.QTY_20)
+        );
+
+        utils = await UtilsBuilder.create()
+          .ERC20withPermit()
+          .TKNTKN()
+          .buildAsync(
+            contractERC1155ERC721,
+            contractVoucherKernel,
+            contractCashier,
+            contractBosonRouter,
+            contractBSNTokenDeposit,
+            contractBSNTokenDeposit
+          );
+
+        await utils.mintTokens(
+          'contractBSNTokenDeposit',
+          users.seller.address,
+          tokensToMint
+        );
+
+        await utils.mintTokens(
+          'contractBSNTokenDeposit',
+          users.buyer.address,
+          tokensToMint
+        );
+
+        await contractERC1155NonTransferable.mint(
+          users.buyer.address,
+          constants.NFT_TOKEN_ID,
+          constants.ONE,
+          constants.ZERO_BYTES
+        );
+
+        const txOrder = await utils.createOrderConditional(
+          users.seller,
+          timestamp,
+          timestamp + constants.SECONDS_IN_DAY,
+          constants.seller_deposit,
+          constants.QTY_10,
+          contractGate,
+          0
+        );
+
+        const txReceipt = await txOrder.wait();
+
+        let eventArgs;
+
+        eventUtils.assertEventEmitted(
+          txReceipt,
+          BosonRouter_Factory,
+          eventNames.LOG_ORDER_CREATED,
+          (e) => (eventArgs = e)
+        );
+
+        tokenSupplyKey = eventArgs._tokenIdSupply;
+
+        await contractGate.registerVoucherSetID(
+          tokenSupplyKey,
+          constants.NFT_TOKEN_ID
+        );
+      });
+
+      async function generateInputs(
```

Comment:
I can work on this.

Reply:
There are three different `generateInputs` functions, each depending on `describe` block, where it is. So the best I can do is just to move before `beforeEach` (instead of after) in that discribe. I put that in next commit.

I think it should be good for the future to decide on some internal standard how to structure tests - also on some other parts we have different style for testing similar things, which is then harder for others to follow.

Reply:
Top of the `describe` is fine. Then it's consistent. Agreed we need to agree on test standard and structure.